awsx/ec2/vpc.ts

```diff
@@ -489,23 +489,19 @@ export class Vpc extends schema.Vpc<VpcData> {
             .output(parsedSpecs?.normalizedSpecs)
             .apply(vpcConverters.toResolvedSubnetSpecOutputs);
 
-    const verifiedSubnetLayout = subnetLayout.apply((sl) => {
+    const verifiedSubnetLayout = pulumi.jsonStringify(subnetLayout).apply(sl => {
```

Comment:
Why does `pulumi.jsonStringify` handle this better than a regular apply?

All it seems to be doing is calling apply on the output: https://github.com/pulumi/pulumi/blob/e34b753f070b2862263f80882ceb88ce69a14272/sdk/nodejs/output.ts#L1170-L1172

Is it that it's wrapping it in `pulumi.output()` again? If so, why do we need double wrapping in output here?

Reply:
Yes, here is the doc for `pulumi.output()`:

```
output takes any Input value and converts it into an Output, deeply unwrapping nested Input values as necessary.

The expected way to use this function is like so:

var transformed = pulumi.output(someVal).apply(unwrapped => {
   // Do whatever you want now.  'unwrapped' will contain no outputs/promises inside
   // here, so you can easily do whatever sort of transformation is most convenient.
});
```

Interestingly, my initial implementation works just fine for a Node.js program (that's how I tested it) but doesn't work for a Python program (provided by Piers). I don't quite understand where this difference comes from, but it must be about the exact shape of the output that we need to log.

Reply:
So that means by doing an `apply` on a `pulumi.output` we're loosing some of the lifting abilities?
My confusing here comes from the fact that we are doing `pulumi.output` in 482-490. It's just that we `apply` on it again.

Reply:
That almost seems like a bug to me, tbh

Reply:
I think it's by design, although it's very confusing and subtle. Note the types of the `apply` argument in each case:

```
// value.apply:
func: (t: { type: pulumi.Output<string>;}) =>
    pulumi.OutputInstance<{type: pulumi.Output<string>;}>

// pulumi.output(value).apply:
(t: pulumi.UnwrappedObject<{  type: pulumi.Output<string>; }>) =>     pulumi.UnwrappedObject<{ type: pulumi.Output<string>; }>

// pulumi.jsonStringify(value).apply:
func: (t: string) => string
```

`UnwrappedObject` is a Mapped Type that removes outputs from properties, so it's plain. But in the first case, you see you get something that isn't safe to serialize.